analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -0,0 +1,316 @@
+            val packageList = removeProjectPackage(rootNode, definitionFile)
+            val packages = extractPackages(packageList, workingDir)
+
+            val dependenciesScope = Scope(
+                name = SCOPE_NAME_DEPENDENCIES,
+                dependencies = extractDependencies(rootNode, SCOPE_NAME_DEPENDENCIES, workingDir)
+            )
+            val devDependenciesScope = Scope(
+                name = SCOPE_NAME_DEV_DEPENDENCIES,
+                dependencies = extractDependencies(rootNode, SCOPE_NAME_DEV_DEPENDENCIES, workingDir)
+            )
+
+            val projectPackage = extractProjectPackage(rootNode, definitionFile, workingDir)
+
+            return ProjectAnalyzerResult(
+                project = Project(
+                    id = projectPackage.id,
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    declaredLicenses = projectPackage.declaredLicenses,
+                    vcs = projectPackage.vcs,
+                    vcsProcessed = processProjectVcs(workingDir, projectPackage.vcs, projectPackage.homepageUrl),
+                    homepageUrl = projectPackage.homepageUrl,
+                    scopes = sortedSetOf(dependenciesScope, devDependenciesScope)
+                ),
+                packages = packages.map { it.value.toCuratedPackage() }.toSortedSet()
+            )
+        }
+    }
+
+    private fun extractDependencyTree(
+        rootNode: JsonNode,
+        workingDir: File,
+        pkg: JsonNode,
+        scopeName: String
+    ): SortedSet<PackageReference> {
+        val result = mutableSetOf<PackageReference>()
+        pkg[ scopeName ]?.forEach {
+            val childRef = it.textValueOrEmpty()
+            rootNode.iterator().forEach { child ->
+                if (child["reference"].textValueOrEmpty() == childRef) {
+                    log.debug { "Found child. '$childRef'" }
+
+                    val packageReference = PackageReference(
+                        id = extractPackageId(child, workingDir),
+                        dependencies = extractDependencyTree(rootNode, workingDir, child, SCOPE_NAME_DEPENDENCIES)
+                    )
+                    result.add(packageReference)
+
+                    val packageDevReference = PackageReference(
+                        id = extractPackageId(child, workingDir),
+                        dependencies = extractDependencyTree(rootNode, workingDir, child, SCOPE_NAME_DEV_DEPENDENCIES)
+                    )
+                    result.add(packageDevReference)
+                }
+            }
+        }
+        return result.toSortedSet()
+    }
+
+    /**
+     * Runs through each package and extracts list of its dependencies (also transitive ones)
+     */
+    private fun extractDependencies(
+        rootNode: JsonNode,
+        scopeName: String,
+        workingDir: File
+    ): SortedSet<PackageReference> {
+        val stack = Stack<JsonNode>()
```

Comment:
Nit: Could do `val stack = Stack<JsonNode>().apply { addAll(rootNode) }` instead of line 173.